### PR TITLE
Specify the target store name to achieve a soup per store

### PR DESCRIPTION
**Pull request by trooper2013**
Changes for ability to specify storeName, addition of Store related API(s).

### CODE REVIEW

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.h [hunk 1]
```diff
@@ -154,6 +154,36 @@ extern NSString * const kSmartStorePluginIdentifier;
  *
  */
 - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command;
+
+/**
+ * Get soup spec details for the given soup name.
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.h]: Looks like copy-paste. Needs to be updated ;-)

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.h [hunk 2]
```diff
@@ -154,6 +154,36 @@ extern NSString * const kSmartStorePluginIdentifier;
...
+ *
+ */
+-(void)pgGetAllGlobalStores:(CDVInvokedUrlCommand *)command;
+
+/**
+ * Get soup spec details for the given soup name.
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.h]: Looks like copy-paste. Needs to be updated ;-)

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/*
+ SFMutableDictionaryThreadSafe.h
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on Wed Oct 21 17:47:00 PDT 2016.
+ 
+ Copyright (c) 2016, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.h]: `2016-present`

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.m [hunk 4]
```diff
@@ -0,0 +1,154 @@
+/*
+ SFMutableDictionaryThreadSafe.m
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on Wed Oct 21 17:47:00 PDT 2016.
+ 
+ Copyright (c) 2016, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.m]: `2016-present`

File: libs/SmartStore/SmartStoreTests/SFMultipleSmartStoresTest.h [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/*
+ SFMultipleSmartStoresTests.h
+ 
+ Created by Raj Rao on Wed Oct 24 17:47:00 PDT 2016.
+ 
+ Copyright (c) 2016, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SmartStore/SmartStoreTests/SFMultipleSmartStoresTest.h]: `2016-present`

File: libs/SmartStore/SmartStoreTests/SFMultipleSmartStoresTests.m [hunk 6]
```diff
@@ -0,0 +1,118 @@
+/*
+ SFMultipleSmartStoresTests.m
+ 
+ Created by Raj Rao on Wed Oct 24 17:47:00 PDT 2016.
+ 
+ Copyright (c) 2016, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline libs/SmartStore/SmartStoreTests/SFMultipleSmartStoresTests.m]: `2016-present`

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
SmartStoreTestSuite | testCreateMultipleGlobalStores|
SmartStoreTestSuite | testCreateMultipleUserStores|
SmartStoreTestSuite | testGetAllGlobalStores|
SmartStoreTestSuite | testGetAllStores|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecords|
SyncManagerTests | testSyncUpWithLocallyCreatedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithLocallyCreatedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithLocallyDeletedRecords|
SyncManagerTests | testSyncUpWithLocallyDeletedRecords|
SyncManagerTests | testSyncUpWithLocallyDeletedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithLocallyDeletedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithLocallyDeletedRemotelyDeletedRecords|
SyncManagerTests | testSyncUpWithLocallyDeletedRemotelyDeletedRecords|
SyncManagerTests | testSyncUpWithLocallyUpdatedRecords|
SyncManagerTests | testSyncUpWithLocallyUpdatedRecords|
SyncManagerTests | testSyncUpWithLocallyUpdatedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithLocallyUpdatedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithLocallyUpdatedRemotelyDeletedRecords|
SyncManagerTests | testSyncUpWithLocallyUpdatedRemotelyDeletedRecords|
SyncManagerTests | testSyncUpWithLocallyUpdatedRemotelyDeletedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithLocallyUpdatedRemotelyDeletedRecordsWithoutOverwrite|
SyncManagerTests | testSyncUpWithNoLocalUpdates|
SyncManagerTests | testSyncUpWithNoLocalUpdates|


## Current coverage for SalesforceAnalytics.framework is `73%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCore.framework is `51%`
Files changed | - | - 
--- | --- | ---
SFMutableDictionaryThreadSafe.m | `45%` | :no_entry_sign:
SFMutableDictionaryThreadSafe.m | `45%` | :no_entry_sign:

---
## Current coverage for CocoaLumberjack.framework is `32%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartStore.framework is `63%`
Files changed | - | - 
--- | --- | ---
SFSmartStore.m | `88%` | :white_check_mark:
SFSmartStore.m | `88%` | :white_check_mark:

---
## Current coverage for SmartSync.framework is `76%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceHybridSDK.framework is `42%`
Files changed | - | - 
--- | --- | ---
SFSmartStorePlugin.m | `73%` | :warning:
SFSmartStorePlugin.m | `73%` | :warning:
SFSmartSyncPlugin.m | `89%` | :white_check_mark:
SFSmartSyncPlugin.m | `89%` | :white_check_mark:

---
## Current coverage for SalesforceReact.framework is `0%`
✅ *No files affecting coverage found*

---
## Current coverage for RestAPIExplorer.app is `26%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartSyncExplorer.app is `19%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartSyncExplorerCommon.framework is `0%`
✅ *No files affecting coverage found*

---
## Current coverage for RecentContactsTodayExtension.appex is `0%`
✅ *No files affecting coverage found*

---
## Current coverage for AccountEditor.app is `0%`
✅ *No files affecting coverage found*

---
## Current coverage for NoteSync.app is `0%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartSyncExplorerHybrid.app is `0%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceSDKCoreTestApp.app is `30%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartStoreTestApp.app is `61%`
✅ *No files affecting coverage found*

---
## Current coverage for SmartSyncTestApp.app is `61%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceHybridSDKTestApp.app is `85%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceAnalyticsTestApp.app is `61%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.h [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+/**
+ This class provides a thread safe Mutable Dictionary. Allows for concurrent reads and exclusive writes.
+ Employs the use of dispatch_sync for reads and dispatch_barrier_async for writes to an
+ internally managed NSMutableDictionary. dispatch_barrier_async() introduces a synchronization
+ point allowing previously enqueued blocks to complete. All blocks placed after the barrier will
+ execute oinly after the barrier block is completed.
+ */
+#import <Foundation/Foundation.h>
+
+@interface SFMutableDictionaryThreadSafe : NSObject
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.h]: I haven't seen a threshold of thread safety concerns to warrant such a wrapper class.  Everywhere this is used:

1. Does atomic actions (i.e. no iteration over keys, values)
2. Doesn't otherwise provide much in the way of one call clashing with another.

If we really feel that we must ensure the thread safety of these operations, I'd propose simply creating a serial dispatch queue in `SFSmartStorePlugin`, and `dispatch_sync` the calls that require exclusive access.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 8]
```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
...
 
 - (BOOL)isGlobal:(NSDictionary *)args
 {
     return args[kIsGlobalStoreArg] != nil && [args[kIsGlobalStoreArg] boolValue];
 }
 
+- (NSString *)storeName:(NSDictionary *)args
+{
+    NSString *storeName = args[kStoreName];
+    if(storeName==NULL || storeName == (id)[NSNull null] || [storeName stringByTrimmingCharactersInSet:
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: This seems like a good candidate for `[args nonNullObjectForKey:]`.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 9]
```diff
@@ -51,63 +52,48 @@
...
-
-- (SFSmartStore *)store
-{
-    return [SFSmartStore sharedStoreWithName:kDefaultSmartStoreName];
-}
-
-- (SFSmartStore *)globalStore
-{
-    return [SFSmartStore sharedGlobalStoreWithName:kDefaultSmartStoreName];
+    [[self cursorCache] removeAllObjects];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: Let's use the `@property` notation: `self.cursorCache`.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 10]
```diff
@@ -51,63 +52,48 @@
...
-    self.userCursorCache = [[NSMutableDictionary alloc] init];
-    self.globalCursorCache = [[NSMutableDictionary alloc] init];
-    self.inspector = [[SFSmartStoreInspectorViewController alloc] initWithStore:self.store];
-    self.globalInspector = [[SFSmartStoreInspectorViewController alloc] initWithStore:self.globalStore];
+    self.cursorCache = [[SFMutableDictionaryThreadSafe alloc] init];
+    self.databaseInspectors = [[SFMutableDictionaryThreadSafe alloc] init];
 }
 
 #pragma mark - Object bridging helpers
+- (SFStoreCursor*)cursorByCursorId:(NSString*)cursorId isGlobal:(BOOL)isGlobal {
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: I'm having a hard time articulating what I want to be different here, specifically.  Generally, this cursor ID pattern is hard to follow, which is going to make it hard to maintain.

My suggestion is to pull the internal cursor ID stuff out into its own object, and then consolidate the accessor functionality in this class accordingly.  There are simply too many methods and parameters flying around here, that do nearly the same thing.  It's hard to follow.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 11]
```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
     } command:command];
 }
 
+- (void)pgGetAllGlobalStores:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        NSArray *allStoreNames = [SFSmartStore allGlobalStoreNames];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: This should be a `copy` of the array, to avoid threading issues during iteration.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 12]
```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
...
+        NSMutableArray *result = [NSMutableArray array];
+        if (allStoreNames.count >0 ) {
+            for(NSString *storeName in allStoreNames) {
+                NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+                [dictionary setValue:storeName forKey:kStoreName];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: Please use modern Objective-C syntax in the following lines: e.g. `dictionary[kStoreName] = storeName;`

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 13]
```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
...
+            }
+        }
+        return [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:result];
+     }
+     command:command];
+}
+
+- (void)pgGetAllStores:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        NSArray *allStoreNames = [SFSmartStore allStoreNames];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: Same as above : `copy` this array.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 14]
```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
...
+        NSMutableArray *result = [NSMutableArray array];
+        if (allStoreNames.count >0 ) {
+            for(NSString *storeName in allStoreNames) {
+                NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+                [dictionary setValue:storeName forKey:kStoreName];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: Same as above: modern Objective-C syntax.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 15]
```diff
@@ -387,14 +368,139 @@ - (void)pgGetSoupSpec:(CDVInvokedUrlCommand *)command {
...
+           
+        }
+        return [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsArray:result];
+    } command:command];
+
+}
+
+- (void)pgRemoveStore:(CDVInvokedUrlCommand *)command {
+    [self runCommand:^(NSDictionary* argsDict) {
+        BOOL isGlobal = [self isGlobal:argsDict];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: This is another area where, similar to the cursor stuff, I wonder if it would be valuable to extract the store name and global flag into a class or struct, to consolidate/encapsulate the helper methods around it, and clean up this class.  It's turning into a sort of functional spaghetti at this point.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 16]
```diff
@@ -51,63 +52,48 @@
...
 @interface SFSmartStorePlugin() 
 
-@property (nonatomic, strong) SFSmartStoreInspectorViewController *inspector;
-@property (nonatomic, strong) SFSmartStoreInspectorViewController *globalInspector;
-@property (nonatomic, strong) SFSmartStore *store;
-@property (nonatomic, strong) SFSmartStore *globalStore;
-@property (nonatomic, strong) NSMutableDictionary *userCursorCache;
-@property (nonatomic, strong) NSMutableDictionary *globalCursorCache;
-
+@property (nonatomic, strong) SFMutableDictionaryThreadSafe *databaseInspectors;
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: This is the continuation of an existing anti-pattern that I'd prefer to see squashed.  This idea of persisted inspector view controllers is unnecessary—the persistence of the view controller should be managed by the view state.  I.e. `self.viewController` presents it, it exists.  `self.viewController` dismisses it, it's reclaimed.  And the rest of the state of it should exist as properties inside the view controller class itself.

Can we clean this up?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.h (hunk 7 above)
Comment (trooper2013) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.h]: I agree that the wrapper may not be required. The SFSmartStorePlugin.m seems to have   mutable state. IMO it might be better to wrap it into a dispatch_sync. Will refactor code to move to serial-Q locally.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m (hunk 11 above)
Comment (trooper2013) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: The underlying implementation returns a  new copy for every call. Do we need a copy?

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m (hunk 15 above)
Comment (trooper2013) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: The underlying smartstore API(s) define different static methods to invoke based on whether the store is global or not. There could be value to adding a SFStoreConfig class only if we were to  add more generic add/remove store apis to the SFSmartStore that are based on this StoreConfig object used as input. The StoreConfig class would at the least have these  2 properties (storeName,isGlobalStore).

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m (hunk 16 above)
Comment (trooper2013) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: Will  clean this one up.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m (hunk 11 above)
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: Good point.  Given that the array is dynamically generated at its source, copying doesn't really buy us anything here.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 17]
```diff
@@ -351,8 +343,7 @@ - (void)pgReIndexSoup:(CDVInvokedUrlCommand *)command
 - (void)pgShowInspector:(CDVInvokedUrlCommand *)command
 {
     [self runCommand:^(NSDictionary* argsDict) {
-        BOOL isGlobal = [self isGlobal:argsDict];
-        SFSmartStoreInspectorViewController* inspector = isGlobal ? self.globalInspector : self.inspector;
+        SFSmartStoreInspectorViewController* inspector = [self inspector:command];
         [inspector present:self.viewController];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: One more request: can we do away with this pattern as well, and the `dismiss` pattern in `SFSmartStoreInspectorViewController`?  They're both substituting for standard view controller presentation patterns:

`[self.viewController presentViewController:inspector animated:NO completion:nil];`

and
```obj-c
- (void) backButtonClicked
{
    [self.presentingViewController dismissViewControllerAnimated:NO completion:NULL];
}
```

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m (hunk 15 above)
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: This all looks a lot better from the consumption side now.  I think the cleanup helped.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 18]
```diff
@@ -343,8 +343,11 @@ - (void)pgReIndexSoup:(CDVInvokedUrlCommand *)command
 - (void)pgShowInspector:(CDVInvokedUrlCommand *)command
 {
     [self runCommand:^(NSDictionary* argsDict) {
-        SFSmartStoreInspectorViewController* inspector = [self inspector:command];
-        [inspector present:self.viewController];
+        __weak typeof(self) weakSelf = self;
+        __block SFSmartStoreInspectorViewController *inspector = [self inspector:command];
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m]: This doesn't need to be `__block`.  Its value doesn't change inside the block where it's used.
